mirdata/beatles.py

```diff
@@ -30,28 +29,45 @@
 
 
 def download(data_home=None, clobber=False):
-    save_path = get_save_path(data_home)
-    download_path = download_from_remote(
+    save_path = utils.get_save_path(data_home)
+    beatles_annotations_path = os.path.join(save_path, BEATLES_DIR)
+
+    if clobber:
+        utils.clobber_all(BEATLES_ANNOT_REMOTE,
+                          beatles_annotations_path,
+                          data_home)
+
+    if utils.check_validated(beatles_annotations_path):
+        print("""
+                The {} dataset has already been downloaded and validated.
+                Skipping download of dataset. If you feel this is a mistake please
+                rerun and set clobber to true
+                """.format(BEATLES_DIR))
+        return
+
+    download_path = utils.download_from_remote(
         BEATLES_ANNOT_REMOTE, data_home=data_home, clobber=clobber)
-    beatles_annotations_path = os.path.join(
-        save_path, BEATLES_DIR, 'annotations')
     if not os.path.exists(beatles_annotations_path):
         os.makedirs(beatles_annotations_path)
-    untar(download_path, beatles_annotations_path, cleanup=True)
-    validate(data_home)
-    print("""
-        Unfortunately the audio files of the Beatles dataset are not available
-        for download. If you have the Beatles dataset, place the contents into
-        a folder called Beatles with the following structure:
-            > Beatles/
-                > annotations/
-                > audio/
-        and copy the Beatles folder to {}
-    """.format(save_path))
+    utils.untar(download_path, beatles_annotations_path, cleanup=True)
+    missing_files, invalid_checksums = validate(data_home)
+    if missing_files or invalid_checksums:
+        print("""
+            Unfortunately the audio files of the Beatles dataset are not available
+            for download. If you have the Beatles dataset, place the contents into
+            a folder called Beatles with the following structure:
+                > Beatles/
+                    > annotations/
+                    > audio/
+            and copy the Beatles folder to {}
+        """.format(save_path))
+        utils.create_invalid(beatles_annotations_path, missing_files, invalid_checksums)
```

Comment:
Can we instead put the `utils.create_invalid` and `utils.create_validated` functions inside `utils.validator`? And `utils.validator` can also first run `utils.check_validated`.

Reply:
I was thinking about this, but I decided not to because I saw extra logic in the beatles dataset with the print statement and thought that if we shouldnt be writing these files if other datasets do extra with the missing files (what if they have retry logic for example?)

I also felt that it was weird for the validator to take upon itself extra work and require both data_home and the annotations path and make it a little bit trickier to test.

However, doing this would save me a lot of time refactoring all the datasets...

Reply:
hmm true. Maybe the print statement can be conditioned on the output of `utils.validator`?

Reply:
Update after in-person conversation. We will move the logic into validator.